### PR TITLE
checkpoints: use TIMEOUT on certain network errors

### PR DESCRIPTION
**Pull request by sumnerevans**

### CODE REVIEW

File: portal.go [hunk 1]
```diff
@@ -895,7 +918,7 @@ func (portal *Portal) HandleMatrixRedaction(evt *event.Event) {
 			portal.bridge.AS.SendMessageSendCheckpoint(evt, appservice.StepRemote, 0)
 		}
 	}
-	portal.bridge.AS.SendErrorMessageSendCheckpoint(evt, appservice.StepRemote, fmt.Errorf("Can't redact non-reaction event"), true, 0)
+	portal.sendUnsupportedCheckpoint(evt, appservice.StepRemote, fmt.Errorf("Can't redact non-redaction event"))
```
Comment (tulir) [inline portal.go:921]: Pretty sure it meant non-reaction events, not non-redaction events

```suggestion
	portal.sendUnsupportedCheckpoint(evt, appservice.StepRemote, fmt.Errorf("Can't redact non-reaction event"))
```

File: portal.go [hunk 2]
```diff
@@ -879,9 +893,18 @@ func (portal *Portal) HandleMatrixReaction(evt *event.Event) {
...
 	}
+
+	if errorMsg != "" {
+		portal.bridge.AS.SendErrorMessageSendCheckpoint(evt, appservice.StepRemote, errors.New(errorMsg), true, 0)
+	}
 }
 
 func (portal *Portal) HandleMatrixRedaction(evt *event.Event) {
+	if !portal.bridge.IM.Capabilities().SendTapbacks {
+		portal.sendUnsupportedCheckpoint(evt, appservice.StepRemote, errors.New("Reaction is not supported in portal"))
```
Comment (tulir) [inline portal.go:904]:
```suggestion
		portal.sendUnsupportedCheckpoint(evt, appservice.StepRemote, errors.New("Bridge does not support any kinds of redactions"))
```

File: portal.go [hunk 3]
```diff
@@ -838,25 +845,32 @@ func (portal *Portal) HandleMatrixMessage(evt *event.Event) {
...
-		errMsg := fmt.Sprintf("Unknown reaction target %s", reaction.RelatesTo.EventID)
-		portal.log.Debugfln(errMsg)
-		portal.bridge.AS.SendErrorMessageSendCheckpoint(evt, appservice.StepRemote, errors.New(errMsg), true, 0)
+		errorMsg = fmt.Sprintf("Unknown reaction target %s", reaction.RelatesTo.EventID)
 	} else if existing := portal.bridge.DB.Tapback.GetByGUID(portal.GUID, target.GUID, target.Part, ""); existing != nil && existing.Type == tapbackType {
-		portal.log.Debugfln("Ignoring outgoing tapback to %s/%s: type is same", reaction.RelatesTo.EventID, target.GUID)
+		errorMsg = fmt.Sprintf("Ignoring outgoing tapback to %s/%s: type is same", reaction.RelatesTo.EventID, target.GUID)
 	} else if resp, err := portal.bridge.IM.SendTapback(portal.GUID, target.GUID, target.Part, tapbackType, false); err != nil {
-		errMsg := fmt.Sprintf("Failed to send tapback %d to %s: %v", tapbackType, target.GUID, err)
-		portal.log.Errorfln(errMsg)
```
Comment (tulir) [inline portal.go:858]: These logs shouldn't be removed

File: portal.go [hunk 4]
```diff
@@ -838,25 +845,32 @@ func (portal *Portal) HandleMatrixMessage(evt *event.Event) {
...
+}
+
 func (portal *Portal) HandleMatrixReaction(evt *event.Event) {
 	if !portal.bridge.IM.Capabilities().SendTapbacks {
+		portal.sendUnsupportedCheckpoint(evt, appservice.StepRemote, errors.New("Reaction is not supported in portal"))
 		return
 	}
 	portal.log.Debugln("Starting handling of Matrix reaction", evt.ID)
+
+	errorMsg := ""
```
Comment (tulir) [inline portal.go:862]:
```suggestion
	var errorMsg string
```

File: portal.go (hunk 2 above)
Comment (sumnerevans) [reply portal.go:904]: This isn't true, though... it supports them in certain conversations.

**Review by tulir — APPROVED** [review]

**Pull request merged**